wom/src/main/scala/wdl4s/wom/graph/GraphNode.scala

```diff
@@ -77,20 +79,28 @@ object GraphNode {
     * @return The new input port, and a GraphInputNode if we needed to make one. Wrapped in a case class.
     */
   private[graph] def linkInputPort(inputPrefix: String, inputMapping: Map[String, OutputPort], callNodeRef: Unit => GraphNode)(inputDefinition: Callable.InputDefinition): LinkedInputPort = {
-    val (outputPort, graphNode): (OutputPort, Option[GraphInputNode]) = inputDefinition match {
-      case input if inputMapping.contains(input.name) => (inputMapping(input.name), None)
+
+    def withInputPort(outputPort: OutputPort, graphNode: Option[GraphInputNode]) = {
+      val inputPort = ConnectedInputPort(inputDefinition.name, inputDefinition.womType, outputPort, callNodeRef)
+      LinkedInputPort(Option(inputPort), outputPort, graphNode)
+    }
+
+    // Unsatisfied GraphInputDefinitions generate ExternalGraphInputNodes
+    inputDefinition match {
+      case input if inputMapping.contains(input.name) => withInputPort(inputMapping(input.name), None)
```

Comment:
dumb question: what is this case?

Reply:
I don't think this can happen from a WDL workflow since everything is an expression but in CWL, before calling `callWithInputs` we actively try to map all inputs of the step to other step outputs or workflow inputs. So we actually expect the outputPort to be in this `inputMapping` most of the time. If it isn't then we need a new GraphInputNode.